pex/dependency_manager.py

```diff
@@ -0,0 +1,108 @@
+# Copyright 2023 Pants project contributors (see CONTRIBUTORS.md).
+# Licensed under the Apache License, Version 2.0 (see LICENSE).
+
+from __future__ import absolute_import
+
+from collections import defaultdict
+
+from pex import pex_warnings
+from pex.dist_metadata import Requirement
+from pex.environment import PEXEnvironment
+from pex.exclude_configuration import ExcludeConfiguration
+from pex.fingerprinted_distribution import FingerprintedDistribution
+from pex.orderedset import OrderedSet
+from pex.pep_503 import ProjectName
+from pex.pex_builder import PEXBuilder
+from pex.pex_info import PexInfo
+from pex.resolve.resolvers import Installed
+from pex.tracer import TRACER
+from pex.typing import TYPE_CHECKING
+
+if TYPE_CHECKING:
+    from typing import DefaultDict, Iterable, Iterator
+
+    import attr  # vendor:skip
+else:
+    from pex.third_party import attr
+
+
+@attr.s
+class DependencyManager(object):
+    _requirements = attr.ib(factory=OrderedSet)  # type: OrderedSet[Requirement]
+    _distributions = attr.ib(factory=OrderedSet)  # type: OrderedSet[FingerprintedDistribution]
+
+    def add_from_pex(self, pex):
```

Comment:
I notice that this function loses any exclusions in `pex`. Is that correct?

It seems like this might lead to attempting to find some dependency that isn't available, because its required by one of this PEX's requirements, but has been excluded?

Reply:
Yes, I think you've found a bug here.

Reply:
Ok, added a new integration test that fails without the fix. Thanks for catching that.